crates/ruff_notebook/src/notebook.rs

```diff
@@ -80,14 +80,127 @@ impl Cell {
         // Ignore cells containing cell magic as they act on the entire cell
         // as compared to line magic which acts on a single line.
         !match source {
-            SourceValue::String(string) => string
-                .lines()
-                .any(|line| line.trim_start().starts_with("%%")),
+            SourceValue::String(string) => string.lines().any(Self::is_magic_cell),
             SourceValue::StringArray(string_array) => string_array
                 .iter()
-                .any(|line| line.trim_start().starts_with("%%")),
+                .map(String::as_str)
+                .any(Self::is_magic_cell),
         }
     }
+
+    /// Returns `true` if a cell should be ignored due to the use of cell magics.
+    fn is_magic_cell(line: &str) -> bool {
+        let line = line.trim_start();
```

Comment:
Can we still trim, but only test the first line, rather than all subsequent lines?

Reply:
Hmm, we actually run the risk of some false positives here...

For example, I guess we'd now flag `alias + 1` as an automagic, incorrectly.

We may need to try parsing this cell...? And fall back to automagics?

Reply:
> Can we still trim, but only test the first line, rather than all subsequent lines?

That might risk in not detecting cases where there's Python code _before_ the magic commands.

> For example, I guess we'd now flag `alias + 1` as an automagic, incorrectly.

Are you referring `alias` as a variable name? Yes, that's basically a risk for all escape commands 😅 

> We may need to try parsing this cell...? And fall back to automagics?

How would this help?